streamlit_drawable_canvas/frontend/src/lib/rect.ts

```diff
@@ -71,8 +73,13 @@ class RectTool extends FabricTool {
     if (this.currentStartY > pointer.y) {
       this.currentRect.set({ top: Math.abs(pointer.y) })
     }
-    this.currentRect.set({ width: Math.abs(this.currentStartX - pointer.x) })
-    this.currentRect.set({ height: Math.abs(this.currentStartY - pointer.y) })
+    let minSize = this.minSize
+    let _width = Math.abs(this.currentStartX - pointer.x);
+    let _height = Math.abs(this.currentStartY - pointer.y);
+    if (_width <= minSize) { _width = minSize }
+    if (_height <= minSize) { _height = minSize }
+    this.currentRect.set({ width: _width })
+    this.currentRect.set({ height: _height })
```

Comment:
Same as above

Could you do it more declaratively with `this.currentRect.set({ width: Math.max(_width, this.minSize) })` instead?

* Also I think there is no need for `let minSize = this.minSize`, the code is simple enough to know that we can't inadvertedly mutate the value :)